tests/testsuite/build.rs

```diff
@@ -2154,6 +2154,56 @@ fn crate_library_path_env_var() {
     setenv_for_removing_empty_component(p.cargo("run")).run();
 }
 
+// See https://github.com/rust-lang/cargo/issues/14194
+#[cargo_test]
+fn issue_14194_deduplicate_library_path_env_var() {
+    let p = project()
+        .file(
+            "src/main.rs",
+            &format!(
+                r#"
+                    use std::process::Command;
+                    fn main() {{
+                        let args: Vec<String> = std::env::args().collect();
+                        if args.len() == 1 {{
+                            return;
+                        }}
+                        let level: i32  = args[1].parse().unwrap();
+                        if level == 0 {{
+                            let lib_path = std::env::var("{}").unwrap();
+                            let paths: Vec<_> = std::env::split_paths(&lib_path).collect();
+                            if paths.len() > 1 {{
+                                let first = &paths[0];
+                                assert!(!paths[1..].contains(&first));
+                            }}
+                            return; 
+                        }} 
+                        
+                        let _  = Command::new(std::env!("CARGO"))
+                        .arg("run")
+                        .arg("--")
+                        .arg((level - 1).to_string())
+                        .status()
+                        .unwrap();
+                    }}
+                "#,
+                dylib_path_envvar(),
+            ),
+        )
+        .build();
+
+    setenv_for_removing_empty_component(p.cargo("run -- 1"))
```

Comment:
I would suggest 3, so it can assure env var stay the same for more than one recursive call.

```suggestion
    setenv_for_removing_empty_component(p.cargo("run -- 3"))
```